src/MICore/UnixUtilities.cs

```diff
@@ -26,19 +26,26 @@ public static class UnixUtilities
         /// signal handler for SIGHUP on the console (executing the two rm commands)
         /// </summary>
         /// <param name="debuggeeDir">Path to the debuggee directory</param>
-        /// <param name="debuggerCmd">Command to the debugger</param>
         /// <param name="dbgStdInName">File where the stdin for the debugger process is redirected to</param>
         /// <param name="dbgStdOutName">File where the stdout for the debugger process is redirected to</param>
+        /// <param name="pidFifo">File where the debugger pid is written to</param>
+        /// <param name="debuggerCmd">Command to the debugger</param>
         /// <returns></returns>
-        internal static string LaunchLocalDebuggerCommand(string debuggeeDir, string debuggerCmd, string dbgStdInName, string dbgStdOutName)
+        internal static string LaunchLocalDebuggerCommand(
+            string debuggeeDir,
+            string dbgStdInName,
+            string dbgStdOutName,
+            string pidFifo,
+            string debuggerCmd)
         {
-             return string.Format(CultureInfo.InvariantCulture,
-                "cd {0}; DbgTerm=`tty`; trap 'rm {2}; rm {3}' EXIT; {1} --interpreter=mi --tty=$DbgTerm < {2} > {3};",
-                debuggeeDir,
-                debuggerCmd,
-                dbgStdInName,
-                dbgStdOutName
-                );
+            return string.Format(CultureInfo.InvariantCulture,
+               "cd {0}; DbgTerm=`tty`; trap 'rm {2}; rm {3}; rm {4}; rm {5}' EXIT; {6} --interpreter=mi --tty=$DbgTerm < {3} > {4} & pid=$! && echo $pid > {5}; wait $pid; ",
```

Comment:
It looks like you removed {1} and added a 4/5/6 but I don't see additional parameters being passed to String.Format().